### PR TITLE
AG-12847 - fix multiple beans with same name in different modules

### PR DESCRIPTION
**Pull request by peterjrreynolds**

### CODE REVIEW

File: packages/ag-grid-community/src/grid.ts [hunk 1]
```diff
@@ -197,50 +199,22 @@ export class GridCoreCreator {
...
-        const moduleUserComps: ComponentMeta[] = this.extractModuleEntity<ComponentMeta>(registeredModules, (module) =>
-            module.userComponents ? module.userComponents : []
-        );
-
         const registry = context.getBean('userComponentRegistry');
-        moduleUserComps.forEach(({ name, classImp, params }) => {
-            registry.registerDefaultComponent(name, classImp, params);
+        registeredModules.forEach((module) => {
+            if (module.userComponents) {
+                module.userComponents.forEach(({ name, classImp, params }) => {
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/grid.ts]: prefer optional chaining.

File: packages/ag-grid-community/src/main-umd-noStyles.ts [hunk 2]
```diff
@@ -9,11 +9,9 @@ import {
     ModuleRegistry,
 } from './main';
 
-ModuleRegistry.__registerModules(
+ModuleRegistry.registerModules(
     // [CommunityFeaturesModule, ClientSideRowModelModule],
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/main-umd-noStyles.ts]: Remove comments

**Review by StephenCooper — APPROVED** [review]
Happy with the changes

File: packages/ag-grid-community/src/grid.ts (hunk 1 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/grid.ts]: Done

File: packages/ag-grid-community/src/main-umd-noStyles.ts (hunk 2 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/main-umd-noStyles.ts]: Done

**Pull request merged**